src/teleop/jetson/arm_trajectory_server.py

```diff
@@ -30,6 +31,41 @@ def joint_states_callback(msg: JointState):
         joint_states = msg
 
 
+def euclidean_error(threshold: int, feedback: FollowJointTrajectoryFeedback) -> Tuple[bool, str]:
+    position_errors = np.array(feedback.error.positions)
+    error = np.linalg.norm(position_errors)
+
+    if error > threshold:
+        error_msg = f"Euclidean error of {error} exceeded threshold of {threshold}"
+        return True, error_msg
+    return False, ""
+
+
+def joint_error(thresholds: list, feedback: FollowJointTrajectoryFeedback) -> Tuple[bool, str]:
+    position_errors = feedback.error.positions
+
+    for i in range(len(thresholds)):
+        if position_errors[i] > thresholds[i]:
+            error_msg = f"""
+                Joint {chr(i+65)} exceeded error threshold of {thresholds[i]}
+                Expected: {feedback.desired.positions[i]} rad")
+                Actual: {feedback.actual.positions[i]} rad")
+            """
+            return True, error_msg
+    return False, ""
+
+
+# Return true if arm has exceeded the error thresholds
+def error_threshold_exceeded(feedback: FollowJointTrajectoryFeedback) -> Tuple[bool, str]:
+    euclidean_error_threshold = rospy.get_param("teleop/euclidean_error_threshold")
```

Comment:
These functions should probably be class functions, that way you can call rospy.get_param only once instead of getting it from the param server every time you run these error checking functions.

Reply:
I put them outside the class to unit test them.

Reply:
Since they're only being used here, I guess it's fine to leave. Definitely seems like we should be able to figure out a way to unit test class functions though